src/gui/print_widget.cpp

```diff
@@ -1222,20 +1222,16 @@ void PrintWidget::exportToImage()
 const void PrintWidget::exportWorldFile(QString path, qreal pixel_per_mm)
 {
 	const auto& georef = map->getGeoreferencing();
+	const auto& ref_transform = georef.mapToProjected();
 
 	const auto top_left     = georef.toProjectedCoords(MapCoord{map_printer->getPrintArea().topLeft()});
-	const auto bottom_right = georef.toProjectedCoords(MapCoord{map_printer->getPrintArea().bottomRight()});
-	const auto extent = QRectF{top_left, bottom_right};
-
-	// Calulate distance per pixel
-	const auto& page_format = map_printer->getPageFormat();
-	int w = qCeil(page_format.paper_dimensions.width() * pixel_per_mm);
-	int h = qCeil(page_format.paper_dimensions.height() * pixel_per_mm);
-	const auto xscale = extent.width() / w;
-	const auto yscale = extent.height() / h;
-
-	// Not sure that the rotation value is right here but it looks correct in my tests
-	QTransform final_wld(xscale, 0, 0, 0, yscale, 0, top_left.x(), top_left.y());
+
+	const auto xscale = ref_transform.m11() / pixel_per_mm;
+	const auto yscale = ref_transform.m22() / pixel_per_mm;
+	const auto xskew = ref_transform.m12() / pixel_per_mm ;
+	const auto yskew = ref_transform.m21() / pixel_per_mm ;
```

Comment:
The export dialog allows setting _image DPI_ and _map scale_. Both parameters must be taken into account when calculating `pixels_per_mm`.

Quick fix:
```
diff --git a/src/gui/print_widget.cpp b/src/gui/print_widget.cpp
index e2df9fa99d3b..0bfb25d6b5c9 100644
--- a/src/gui/print_widget.cpp
+++ b/src/gui/print_widget.cpp
@@ -1225,11 +1225,12 @@ const void PrintWidget::exportWorldFile(QString path, qreal pixel_per_mm)
        const auto& ref_transform = georef.mapToProjected();
 
        const auto top_left     = georef.toProjectedCoords(MapCoord{map_printer->getPrintArea().topLeft()});
+       const auto pixel_per_mm_adjusted = pixel_per_mm * map_printer->getScaleAdjustment();
 
-       const auto xscale = ref_transform.m11() / pixel_per_mm;
-       const auto yscale = ref_transform.m22() / pixel_per_mm;
-       const auto xskew = ref_transform.m12() / pixel_per_mm ;
-       const auto yskew = ref_transform.m21() / pixel_per_mm ;
+       const auto xscale = ref_transform.m11() / pixel_per_mm_adjusted;
+       const auto yscale = ref_transform.m22() / pixel_per_mm_adjusted;
+       const auto xskew = ref_transform.m12() / pixel_per_mm_adjusted;
+       const auto yskew = ref_transform.m21() / pixel_per_mm_adjusted;
 
        QTransform final_wld(xscale, yskew, 0, xskew, yscale, 0, top_left.x(), top_left.y());
        WorldFile wld(final_wld);
```

Given that `exportWorldFile()` method has access to `map_printer`, the adjusted `pixels_per_mm` should IMHO be calculated in situ and the method parameter should be dropped.

Reply:
It seemed redundant to calculate `pixels_per_mm` again as it is calculated already in the function where `exportWorldFile` is called. I didn't want to duplicate the code, especially as I didn't understand where the magic number 25.4 comes from.

Reply:
Print resolution is in Dots-Per-Inch and an inch has 25.4 mm. :)  Obviously, the constant needs a name.